src/towncrier/test/test_settings.py

```diff
@@ -199,3 +201,70 @@ def test_missing_template_in_towncrier(self):
         self.assertEqual(
             str(e.exception), "Towncrier does not have a template named 'foo'."
         )
+
+    def test_custom_types_as_tables_array(self):
+        """
+        Custom fragment categories can be defined inside
+        the toml config file using an array of tables
+        (a table name in double brackets).
+        """
+        toml_content = """
+        [tool.towncrier]
+        package = "foobar"
+        [[tool.towncrier.type]]
+        directory="foo"
+        name="Foo"
+        showcontent=false
+
+        [[tool.towncrier.type]]
+        directory="spam"
+        name="Spam"
+        showcontent=true
+        """
+        toml_content = textwrap.dedent(toml_content)
+        expected = [
+            (u"foo", {"name": u"Foo", "showcontent": False, }),
+            (u"spam", {"name": u"Spam", "showcontent": True, },),
+        ]
+        expected = clt.OrderedDict(expected)
+        config = self._get_actual_config(toml_content,)
+        actual = config["types"]
+        self.assertDictEqual(expected, actual)
+
+    def test_custom_types_as_tables(self):
+        """
+        Custom fragment categories can be defined inside
+        the toml config file using tables.
+        """
+        test_project_path = self.mktemp()
+        os.makedirs(test_project_path)
+        toml_content = """
+        [tool.towncrier]
+        package = "foobar"
+        [tool.towncrier.fragment.feat]
+        ignored_field="Bazz"
+        [tool.towncrier.fragment.fix]
+        [tool.towncrier.fragment.chore]
+        name = "Other Tasks"
+        showcontent = false
+        """
+        toml_content = textwrap.dedent(toml_content)
+        expected = [
+            (u"chore", {"name": u"Other Tasks", "showcontent": False, }),
+            (u"feat", {"name": u"Feat", "showcontent": True, }),
+            (u"fix", {"name": u"Fix", "showcontent": True, }),
+        ]
+
+        expected = clt.OrderedDict(expected)
+        config = self._get_actual_config(toml_content,)
+        actual = config["types"]
+        self.assertDictEqual(expected, actual)
+
+    def _get_actual_config(self, toml_content):
+        test_project_path = self.mktemp()
```

Comment:
I think there is no need to make it private, as this is testing code.

I think that if a helper method can be used by multiple tests, it can have a public name.

We can still use private names in testing code, for code that should only be used by a single test.

```suggestion
    def load_config(self, toml_content):
        """
        Load the configuration based on a temporary
        file having `toml_content` as content.
        """
        test_project_path = self.mktemp()
```

or make `def config_from_text` or `def load_content` 

any name is ok, but it's important to have a docsting describing the purpose / intent of this helper method.